### PR TITLE
Added max_auto_feedback_points for code question along with tests.

### PR DESCRIPTION
**Pull request by dzhuang**
Patching #448 

Introduced new attribute `max_auto_feedback_points` for code questions.

### CODE REVIEW

**Comment by dzhuang** [comment]
@adamjstewart

**Comment by dzhuang** [comment]
The tests on Travis-CI failed, but passed on Appveyor. I think the reason is the faked subprocess in the tests is not killed after one Testcase which opened that process.

**Comment by dzhuang** [comment]
Confirmed. [this test](https://travis-ci.org/dzhuang/relate/jobs/338392163#L1290) is using the subprocess, and the TestCase followed that failed.

**Comment by dzhuang** [comment]
@adamjstewart I also added test you mentioned in #448 that we should also make sure 0 is tested.
@inducer I am not confident about the naming and docs, can you review that? Thanks.

File: course/page/code.py [hunk 1]
```diff
@@ -426,6 +426,12 @@ class PythonCodeQuestion(PageBaseWithTitle, PageBaseWithValue):
         based on its :attr:`access_rules` (not the ones of the flow), a warning
         is shown. Setting this attribute to True will silence the warning.
 
+    .. attribute:: max_auto_feedback_points
+
+        Optional, A non-negative :class:`float` value, default to 1,
+        and it can be increased to 10 (`course.constants.MAX_EXTRA_CREDIT_FACTOR`).
+        It indicates the maximum allowed correctness for auto feedback.
+
```
Comment (inducer) [inline course/page/code.py]: Do we really need that? What do you think of the simpler solution I proposed in #448 to not validate the point count in the runner and instead just do this in the app server?

Comment (dzhuang) [reply course/page/code.py]: I see this

> Ideally, we'd pass it into the docker container via JSON (as, say, `max_correctness`) and compare with that. 

So I guess that's your prefered solution. Thus I've made it more complex.

Comment (inducer) [reply course/page/code.py]: Yeah. I wrote that, and then last night I realize that I was being an idiot. No need to make it so complicated, IMO. Sorry!

Comment (dzhuang) [reply course/page/code.py]: Nvm, I started out without reading through your later comments. So the work around is remove that validation part in the docker side, and do it in the app server side?

Comment (dzhuang) [reply course/page/code.py]: You are right, It's better to do that in the app server, I'm closing this and open another PR.

**Comment by dzhuang** [comment]
@inducer @adamjstewart When feedback.points falls out of the range of [0, MAX_EXTRA_CREDIT_FACTOR], besides `correctness = None` what is the expected behavior you prefer? 

1. Shall we log an error in the feedback
1. Shall we send an email to the instructor

**Comment by inducer** [comment]
> Shall we log an error in the feedback?

Yeah. I'd suggest treating it the same as "grading code failed".

**Comment by dzhuang** [comment]
Ok, that means we also need to send an email to the instructor. Thanks.

**Pull request closed without merging**